### ISSUE TITLE
Are the images inside the image_log (while training) are validation images or train images?

### ISSUE THREAD

**Issue by HassanBinHaroon** [lllyasviel/ControlNet#416]
Are the images inside the image_log (while training) are **unseen images** or training images?

**Comment by engrmusawarali** [comment]
These are train images, you can generate validation images also by some modifications in logger.py

**Comment by HassanBinHaroon** [comment]
@engrmusawarali Can't we have the validation loss by default by the model?

**Comment by engrmusawarali** [comment]
Yes you can have it, you need to have separate data loader for validation loss

**Comment by HassanBinHaroon** [comment]
@engrmusawarali Did you train it on your own custom data?

**Comment by HassanBinHaroon** [comment]
@engrmusawarali Thanks, Can we have an online session? I need to discuss something when you're available. Kindly send your email if its feasible for you. Thanks!

**Comment by WhiteKnight9** [comment]
@engrmusawarali hi, can i ask a simple question is that after i training my own datasets, it got some images in img_log. But there are four kinds of imges in the img_log folders
![H%B0$QIY 2C 3{ 38%G}TF](https://github.com/lllyasviel/ControlNet/assets/89592450/6db17c26-8fc7-4829-8c67-164dde8d1ef8)
i know the conditionging is the prompt,the control should be source image? but how about samples_cfg and reconstruction? 
sry i ask this question because my source img and target img is almost same only difference is the color style.

**Comment by HassanBinHaroon** [comment]
Control is the source image.

Conditioning is the prompt. 

Reconstruction is the target image. 

Sample is the predicted image which is the result of training.

**Comment by Lanhaoran** [comment]
> @engrmusawarali Thanks, Can we have an online session? I need to discuss something when you're available. Kindly send your email if its feasible for you. Thanks!

Hey @HassanBinHaroon, have you figured out how to enable validation loss for training and log validation images in the folder       'image_log' ? I would be very grateful if you could tell me some！ Thanks in advance !

**Comment by HassanBinHaroon** [comment]
How are you doing @Lanhaoran? 

I didn't try it yet. But I can refer you the link where you can get the insight. The [link](https://lightning.ai/docs/pytorch/stable/common/trainer.html)

And if you do it successfully then please inform it here too. 

BTW, hopefully I'll try it nearly.

**Comment by Lanhaoran** [comment]
> 

OK, I will take a try, thanks for your information !

**Comment by WhiteKnight9** [comment]
> Control is the source image.
> 
> Conditioning is the prompt.
> 
> Reconstruction is the target image.
> 
> Sample is the predicted image which is the result of training.

thanks very much!!

**Comment by HassanBinHaroon** [comment]
@Lanhaoran Thanks for the update!

Actually these links are not working at my side. Can you send the screenshots so I can find my searching? Thanks!

**Comment by Lanhaoran** [comment]
> @Lanhaoran Thanks for the update!
> 
> Actually these links are not working at my side. Can you send the screenshots so I can find my searching? Thanks!

This is the training part in his code.

![1687181191034](https://github.com/lllyasviel/ControlNet/assets/43608881/63f40c81-2016-4bb6-811d-2f34ec4a29b4)
You may try search gabegrand in github and check his repositories for more details. 

Good luck !

**Comment by HassanBinHaroon** [comment]
@Lanhaoran Thanks!

**Comment by Lanhaoran** [comment]
> @Lanhaoran Thanks!

You are welcome !
